### PR TITLE
Adds a bluespace crystal material, balances some R&D recipes around it.

### LINKED ISSUE #23633: Consistency Issue: You can night vision sec and med hugs before night vision goggles
>Plasma 4 
>Emp 5

NVG Sec hud and NVG Med hud

>Get plasma 5
>Suddenly night vision goggles

>Get engi 6 
>Night vision mesons

Highy inconsistent

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: Bluespace Crystals are now a material that can be inserted in Protolathes and Circuit Printers. Some items now require Bluespace Mesh.
add: Bluespace Crystal can now be ground in a reagent grinder to gain bluespace dust. It has no uses, but it teleports people if splashed on them, and if ingested it will occasionally cause teleportation.
/:cl:

<s>Most notable change: AI Upload and Cyborg Upload boards now require Bluespace Mesh, making them require mining to acquire, or at the very least deconstructing the Teleporter.</s>
Other bluespace-themed items now require some Bluespace Mesh, except for a few that were intentionally cheap like the Mining Satchel of Holding.

Balance-wise it's technically rarer than diamonds. Artificial crystals contain no actual mesh, but are still usable for anything requiring whole crystals.

### Other Changes
Since i was editing RnD i lowered basic NV glasses' plasmatech requirement to 4 from 5. Fixes #23633.

Lowers the RnD syringes glass cost to 2000 from 4000 because damn, that's a lot for one syringe. Also lowered diamond cost for the piercing syringe to 1000 from 1500.

### CODE REVIEW

**Comment by GunHog** [comment]
:-1: For the nerf on upload boards, that is a serious nerf!

**Comment by Arianya** [comment]
Why not just add plasma and diamond cost to these items equivalent to their "bluespace mesh" cost. It seems weird to be adding a entirely new "material" as a balancing method when that material is already synthesizeable (and available on station roundstart).

File: code/modules/research/designs.dm [hunk 1]
```diff
@@ -480,7 +480,7 @@ other types of metals and chemistry for reagents).
 	name = "Night Vision Goggles"
 	desc = "Goggles that let you see through darkness unhindered."
 	id = "night_visision_goggles"
-	req_tech = list("materials" = 4, "magnets" = 5, "plasmatech" = 5, "engineering" = 5)
+	req_tech = list("materials" = 4, "magnets" = 5, "plasmatech" = 4, "engineering" = 5)
```
Comment (Arianya) [inline code/modules/research/designs.dm]: Undocumented changes that have nothing to do with bluespace mesh REEEEEEEEE

Comment (XDTM) [inline code/modules/research/designs.dm]: Nuh huh i documented it

**Comment by XDTM** [comment]
It felt like bluespace crystal should be a material in the first place; after all it's now in sheets, collected from mining and smelted along with the rest.

**Comment by XDTM** [comment]
About the upload boards it seems like a fair nerf, considering that giving laws to the AI, a process which would normally require getting into a central, secure, and turreted area, is extremely easy when done in R&D, since all you have to do is AI OPEN the vault for the gold.

**Comment by Arianya** [comment]
I'd argue that if we're going to make it its own mat then you should remove the option to protolathe it out of gold and diamonds. There's no point having a material "rarer then diamonds" that can be made without any real stress out of materials less rare then it.

Also 👍 what Gun said about the upload boards.

**Comment by XDTM** [comment]
Makes sense, i'll make artificial crystals have no materials, so you can still use them for machinery.

**Comment by XDTM** [comment]
I'll revert the upload boards if nobody supports that nerf. I added the cost due to some people criticizing that part of RnD, but i'm fine either way.

**Comment by GunHog** [comment]
This is quite a middle-finger to Science if you remove mats from arti-crystals...

File: code/modules/telesci/bscrystal.dm [hunk 2]
```diff
@@ -48,7 +48,7 @@
 	name = "artificial bluespace crystal"
 	desc = "An artificially made bluespace crystal, it looks delicate."
 	origin_tech = "bluespace=3;plasmatech=4"
-	materials = list(MAT_BLUESPACE=MINERAL_MATERIAL_AMOUNT / 2)
```
Comment (GunHog) [inline code/modules/telesci/bscrystal.dm]: Put the mats back please

**Comment by XDTM** [comment]
Usually you get crystals along with diamonds if miners make it that far, so splitting costs between diamonds and crystals can be a buff since diamonds will last more.

**Comment by GunHog** [comment]
It is not as common as you think, I have seen lots of times where miners do not get crystals, and when they do, it is much, much less than diamonds. It is certainly in no manner a buff.

**Comment by XDTM** [comment]
Yeah, it was intended as sort of a nerf. Science is pretty much the most powerful department after all.

**Comment by PKPenguin321** [comment]
In your changelog,
>but it teleports people is splashed on them

**Comment by XDTM** [comment]
you saw nothing

**Comment by GunHog** [comment]
I would still like a way to convert the diamond and plasma to a small amount of the mats for use in the event that miners fail to get BS crystals.

**Comment by XDTM** [comment]
I re-added it since i realized it makes smelted crystals with low tech when redeemed.

**Comment by GunHog** [comment]
Alrighty, I am okay with this, then!

**Comment by TehZombehz** [comment]
The addition of that bluespace dust will enable me to make fun cooking dishes using bluespace bananas and tomatoes so I can finally do something useless with meme botanists.

**Comment by XDTM** [comment]
Oh shit i forgot i wanted to add it to bluespace plants thanks

**Comment by XDTM** [comment]
makes much more sense than an alcoholic drink anyway

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -807,6 +807,26 @@
...
+/datum/reagent/bluespace
+	name = "Bluespace Dust"
+	id = "bluespace"
+	description = "A dust composed of microscopic bluespace crystals, with minor space-warping properties."
+	reagent_state = SOLID
+	color = "#0000CC"
+
+/datum/reagent/bluespace/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
+	if(method == TOUCH || method == VAPOR)
+		do_teleport(M, get_turf(M), (reac_volume / 10), asoundin = 'sound/effects/phasein.ogg')
```
Comment (kevinz000) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: only 10 tiles for 5 crystals? Kinda weak.

**Comment by kevinz000** [comment]
:+1: on everything else
:-1: for upload boards.

**Comment by GunHog** [comment]
Wait, I thought he reverted that...still against that nerf.

**Comment by PKPenguin321** [comment]
i think it's a fair nerf, i didn't realize R&D could just subvert the AI without minings help, that seems kind of stupid

**Comment by GunHog** [comment]
You still need gold/diamonds for the law boards except reset.

**Comment by Core0verload** [comment]
Remove upload nerf and it'll be a solid 👍

**Comment by MimicFaux** [comment]
Bluespace dust chemical dental pill. Caught by security? Try your luck with a home-made blink!

**Comment by optimumtact** [comment]
Better content gating has been coming for a long time so I dont' know why science players act all suprised.

**Comment by XDTM** [comment]
If there's no support for the upload board nerf i'll revert that part

**Comment by Militaires** [comment]
keep the nerf

sci can subvert roundstart by going to the vault for gold and that's fucking bullshit

**Comment by Cheridan** [comment]
leave it for another pr, please.

**Comment by optimumtact** [comment]
Cheri is right

**Comment by FantasticFwoosh** [comment]
Print bluespace floor tiles out of three quarters metal, plasma and bluespace mesh would be my input.

Bluespace crystals aren't hard to get but it does have a short shelf life for idiots (both accidental & deliberate) who might want to trigger the crystals & get locked in the main next door to science, therefore using poly-crystals would be safer.

Oh & don't forget the RD cloak made out of bluespace mesh that causes them to teleport on extreme trauma & whishing their cape deliberately.

**Comment by XDTM** [comment]
You can insert both single and polycrystals, don't worry. As for new items and recipes it's better to add them separately, i guess.

**Pull request merged**